ckanclient/__init__.py

```diff
@@ -0,0 +1,165 @@
+import logging
+
+import requests
+from frictionless_ckan_mapper import ckan_to_frictionless, frictionless_to_ckan
+
+from ckanclient.auth import CkanAuthApi
+from ckanclient.text import camel_to_snake
+from ckanclient.upload import push_data_to_blob_storage, verify_upload
+
+
+LOG_FORMAT = '%(asctime)s - %(levelname)s - %(message)s'
+logging.basicConfig(format=LOG_FORMAT, level=logging.INFO)
+log = logging.getLogger(__name__)
+
+
+class CkanClientError(Exception):
+    pass
+
+
+class CkanClient:
+    def __init__(self, api_url, api_key, organization, dataset_id, lfs_url):
+        self.api_url = api_url
+        self.api_key = api_key
+        self.organization = organization
+        self.dataset_id = dataset_id
+        self.lfs_url = lfs_url
+        self.auth = CkanAuthApi(api_url, api_key, organization, dataset_id)
+
+    def action(
+        self,
+        name,
+        payload,
+        http_get=False,
+        transform_payload=None,
```

Comment:
i would not have a transform function here. Keep this simpler and clients of this method do the transform.

Reply:
I don't have a great reason other than it seems cleaner and there is an elegance to this so i'm in two minds. For now we can leave this as is since if we did refactor it would not really alter the public API.